### ISSUE TITLE
input source maps only have their first sources considered

### ISSUE THREAD

**Issue by tilgovi** [thlorenz/combine-source-map#8]
At https://github.com/thlorenz/combine-source-map/blob/master/index.js#L30 it's clear that only the first source is considered.

When using combine-source-map to try to take already bundled/transpiled source and add new code, updating the source map, it's helpful to process the mappings for all sources.

See here for an example of using the source-map module directly to port all mappings with a fixed offset: https://github.com/okfn/annotator/blob/browserify/tools/bundle#L145

**Comment by brenmar** [comment]
Any progress with merging the pull request in - or at least explaining why it hasn't been so it can be improved?!

 This bug has been open for over a year and renders browserify source-maps unusable in all but the most basic situations.

**Comment by aarononeal** [comment]
I just submitted a new pull #14 that modifies pull #11 and which I think fully addresses the issue.

**Comment by thlorenz** [comment]
@aarononeal I added you as collaborator.
Now you can merge your proposed PRs into master and ping me when done so I can publish a new version.

Please make sure that the changes don't break anything and still work with the latest browserify.
Also it'd be nice to ensure that new versions get used in new browserify releases, i.e. propose PRs that upgrade their `combine-source-map` dependency.

When you do merge, please do forward merges by rebasing off master first.
I consider `Merge Pull request ...` commits in the history as noise, so I'd like to keep it clean and linear.

Please close any issues once they are resolved due to merged PRs and help out with other issues as you can. I'd greatly appreciate it.

Thanks for your help.

**Comment by aarononeal** [comment]
@thlorenz-- Thanks! I'll tackle that this weekend.

**Comment by aarononeal** [comment]
@thlorenz -- I added an extra test to cover the behavior, added a couple comments for clarity, and FF merged the PR. I ran the change through my Browserify based projects and I'm getting the source paths I would expect. I'll keep an eye on things in case anyone reports a regression.

Once you publish I can send a PR to browser-pack and browersify to update their dependencies. In the meantime, folks can manually add this component to their package.json to get the fix.
